blazar/plugins/oshosts/host_plugin.py

```diff
@@ -973,6 +973,42 @@ def poll_resource_failures(self):
                 recovered_hosts.extend([host for host in unreservable_hosts
                                         if host['id'] in active_hv_ids])
 
+            aggregates = self.nova.aggregates.list()
+            # get all physical hosts
+            hosts = db_api.host_list()
+            pool = nova.ReservationPool()
+            freepool = pool.get_aggregate_from_name_or_id(pool.freepool_name)
+            if not freepool:
+                raise ValueError("Aggregate list does not contain a freepool!")
+            aggregates = self.nova.aggregates.list()
+            # create a map to get the current aggregate of a host in nova
+            host_to_agg_map = {host: agg.name for agg in aggregates for host in agg.hosts}
+            for host in hosts:
+                # get the most recent reservation for the host_id and check if we need to move to freepool
+                reservation = db_utils.get_recent_non_pending_reservation_by_host_id(host.id)
+                # an aggregate not in 'active' or 'pending' reservation should be in freepool
+                if reservation.status == status.reservation.ACTIVE:
```

Comment:
So essentially, a reservation only gets past this point if the reservation is deleted or in error? That seems like correct behavior.